### PR TITLE
Preparation for Open JDK 17

### PR DESCRIPTION
**Pull request by z-sztrom**
### Description of the Change

Necessary changes to compile unser Open JDK 17.

### Benefits

The code isn't limited to old Java version (1.8).

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: 
Roman Szturc roman.szturc.ext@ericsson.com

### CODE REVIEW

File: publish-service/src/main/resources/application.properties [hunk 1]
```diff
@@ -8,7 +8,8 @@ logging.level.org.springframework.web: INFO
 logging.level.com.ericsson.eiffel.remrem.producer: INFO
 
 # Jasypt configurations
-jasypt.encryptor.jasyptKeyFilePath: <The location of the key which was used to encrypt the password, used for decryption of the Encrypted passwords in application >
+# The location of the key which was used to encrypt the password, used for decryption of the Encrypted passwords in application
+jasypt.encryptor.jasyptKeyFilePath: ""
```
Comment (elebroerik) [inline publish-service/src/main/resources/application.properties:12]: Maybe this change will have impact?

File: publish-service/src/main/resources/application.properties [hunk 2]
```diff
@@ -36,3 +37,7 @@ activedirectory.managerDn:
 activedirectory.rootDn:
 activedirectory.userSearchFilter:
 activedirectory.connectionTimeOut:
+
+spring.mvc.pathmatch.matching-strategy=ANT_PATH_MATCHER
```
Comment (elebroerik) [inline publish-service/src/main/resources/application.properties:41]: Do we have Glue ticket adding this property?

Comment (z-sztrom) [reply publish-service/src/main/resources/application.properties:41]: This is necessary due to uplift of spring-boot. Nothing to configure by user.

File: publish-service/src/main/resources/application.properties (hunk 1 above)
Comment (z-sztrom) [reply publish-service/src/main/resources/application.properties]: Good catch! Original value restored.

File: publish-service/pom.xml [hunk 3]
```diff
@@ -119,14 +124,14 @@
             <scope>compile</scope>
         </dependency>
         <dependency>
-            <groupId>com.jayway.restassured</groupId>
+            <groupId>io.rest-assured</groupId>
             <artifactId>rest-assured</artifactId>
             <scope>compile</scope>
```
Comment (vishnu-alapati) [inline publish-service/pom.xml:129]: As i can see the version value is coming from remrem-parent which is 2.9.0 earlier, for groupId you have changed we dont have 2.9.0 version.. see here https://mvnrepository.com/artifact/io.rest-assured/rest-assured

File: publish-service/pom.xml [hunk 4]
```diff
@@ -137,23 +142,33 @@
...
             <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-inline</artifactId>
```
Comment (vishnu-alapati) [inline publish-service/pom.xml:155]: missing the version value... not inheriting from remrem-parent

File: publish-common/pom.xml [hunk 5]
```diff
@@ -133,21 +133,37 @@
...
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.jasypt</groupId>
-            <artifactId>jasypt</artifactId>
-            <version>1.9.2</version>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-inline</artifactId>
+            <version>${mockito.version}</version>
+            <scope>test</scope>
```
Comment (vishnu-alapati) [inline publish-common/pom.xml]: Added twice

File: publish-cli/pom.xml [hunk 6]
```diff
@@ -112,18 +117,22 @@
...
-            <artifactId>powermock-module-junit4</artifactId>
-            <version>${powermock.version}</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.powermock</groupId>
-            <artifactId>powermock-api-mockito2</artifactId>
-            <version>${powermock.version}</version>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-inline</artifactId>
```
Comment (vishnu-alapati) [inline publish-cli/pom.xml:135]: Is it inherit from remrem-parent really? as artifactId is different....

File: publish-common/pom.xml (hunk 5 above)
Comment (z-sztrom) [reply publish-common/pom.xml]: removed

File: publish-cli/pom.xml (hunk 6 above)
Comment (z-sztrom) [reply publish-cli/pom.xml:129]: added ${mockito.version}

File: publish-cli/pom.xml [hunk 7]
```diff
@@ -38,11 +39,13 @@
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-actuator</artifactId>
+            <version>2.7.5</version>
```
Comment (vishnu-alapati) [inline publish-cli/pom.xml:42]: I think we can remove hard coded values, the value is getting inherited from the remrem-parent..

Comment (z-sztrom) [reply publish-cli/pom.xml]: replaced by variable

File: publish-service/src/main/resources/application.properties [hunk 8]
```diff
@@ -5,12 +5,14 @@ debug: false
 spring.mvc.pathmatch.matching-strategy=ANT_PATH_MATCHER
 
 #Logging configurations 
-logging.level.root: INFO
-logging.level.org.springframework.web: INFO
-logging.level.com.ericsson.eiffel.remrem.producer: INFO
+logging.level.root: DEBUG
```
Comment (vishnu-alapati) [inline publish-service/src/main/resources/application.properties:8]: We can change the log-level to info, as this app.prop file is for our local run purpose...

**Review by vishnu-alapati — APPROVED** [review]
Looks fine for me..

File: publish-cli/pom.xml [hunk 9]
```diff
@@ -125,15 +125,9 @@
...
-            <version>${powermock.version}</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.powermock</groupId>
-            <artifactId>powermock-api-mockito2</artifactId>
-            <version>${powermock.version}</version>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-inline</artifactId>
+            <version>${mockito.version}</version>
```
Comment (jainadc9) [inline publish-cli/pom.xml:130]: With mockito 5 inline mockito is supported by default additional configuration for mockmacker inline is not needed if used

File: publish-common/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 10]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (jainadc9) [inline publish-common/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: with mockito 5 this configuration is not needed

File: publish-cli/src/test/java/com/ericsson/eiffel/remrem/publish/cli/CliUnitTests.java [hunk 11]
```diff
@@ -112,44 +101,45 @@ public void testRunCliOptionsMissingFlags() throws Exception {
...
-        PowerMockito.mockStatic(PublishUtils.class);
-        Mockito.when(PublishUtils.getMessageService(Mockito.eq("eiffelsemantics"), Mockito.any()))
-                .thenReturn(eiffelsemanticsMsgService);
-        Mockito.when(eiffelsemanticsMsgService.getServiceName()).thenReturn("eiffelsemantics");
-        Mockito.doThrow(new RemRemPublishException("message")).when(rmqHelper)
-                .rabbitMqPropertiesInit(Mockito.anyString());
-        File file = new File("src/test/resources/publishMessages.json");
-        if (file.exists()) {
-            System.out.println("fileExist");
+        try (MockedStatic<PublishUtils> utils = Mockito.mockStatic(PublishUtils.class)) {
```
Comment (jainadc9) [inline publish-cli/src/test/java/com/ericsson/eiffel/remrem/publish/cli/CliUnitTests.java:104]: Repetitive code better to initialize this before each test using @Before annotation and used elsewhere

**Review by jainadc9 — CHANGES_REQUESTED** [review]

File: publish-service/src/main/resources/application.properties (hunk 8 above)
Comment (z-sztrom) [reply publish-service/src/main/resources/application.properties:8]: Done

File: publish-cli/pom.xml (hunk 9 above)
Comment (z-sztrom) [reply publish-cli/pom.xml:130]: Uplifted to Mockito 5.3.1. Removed mockito-inline.

File: publish-common/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker (hunk 10 above)
Comment (z-sztrom) [reply publish-common/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: Uplifted to Mockito 5.3.1; removed mockito-inline.

File: publish-cli/src/test/java/com/ericsson/eiffel/remrem/publish/cli/CliUnitTests.java (hunk 11 above)
Comment (z-sztrom) [reply publish-cli/src/test/java/com/ericsson/eiffel/remrem/publish/cli/CliUnitTests.java:104]: Moved to setUp().

File: publish-cli/src/test/java/com/ericsson/eiffel/remrem/publish/cli/CliUnitTests.java [hunk 12]
```diff
@@ -112,44 +101,45 @@ public void testRunCliOptionsMissingFlags() throws Exception {
...
-        Mockito.when(messageService.send(Mockito.anyString(), Mockito.any(), Mockito.isNull(), Mockito.isNull(),
-                Mockito.isNull())).thenReturn(res);
-        JsonArray jarray = new JsonArray();
-        List<PublishResultItem> results = new ArrayList<>();
-        jarray.add(results.add(resultItem));
-        Mockito.when(res.getEvents()).thenReturn(results);
-        File file = new File("src/test/resources/publishMessages.json");
-        if (file.exists()) {
-            System.out.println("fileExist");
+        try (MockedStatic<PublishUtils> utils = Mockito.mockStatic(PublishUtils.class)) {
```
[the remark this answers is not in the document]
Comment (z-sztrom) [reply publish-cli/src/test/java/com/ericsson/eiffel/remrem/publish/cli/CliUnitTests.java:123]: Moved to setUp().

File: publish-common/src/test/java/com/ericsson/eiffel/remrem/publish/helper/RMQHelperUnitTest.java [hunk 13]
```diff
@@ -73,15 +68,23 @@ public class RMQHelperUnitTest {
...
-        Mockito.doNothing().when(factory).useSslProtocol();
         Mockito.when(factory.newConnection()).thenReturn(mockConnection);
         Mockito.when(mockConnection.createChannel()).thenReturn(mockChannel);
-        PowerMockito.whenNew(RabbitMqProperties.class).withNoArguments().thenReturn(rabbitMqProperties);
-        initProperties();
-        rabbitMqProperties.setFactory(factory);
-        rmqHelper.rabbitMqPropertiesInit(protocol);
-        rabbitmqProtocolProperties = rmqHelper.getRabbitMqPropertiesMap().get(protocol);
+
+        try (MockedConstruction<RabbitMqProperties> construction = mockConstruction(RabbitMqProperties.class,
```
[the remark this answers is not in the document]
Comment (z-sztrom) [reply publish-common/src/test/java/com/ericsson/eiffel/remrem/publish/helper/RMQHelperUnitTest.java:74]: Hmm, this is in setUp() marked as @Before.

File: publish-service/pom.xml (hunk 3 above)
Comment (z-sztrom) [reply publish-service/pom.xml:123]: rest-assured uses release 5.3.0 now

File: publish-cli/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 14]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (jainadc9) [inline publish-cli/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: Please remove this file not needed with the mockito uplift

**Review by jainadc9 — CHANGES_REQUESTED** [review]

File: publish-cli/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker (hunk 14 above)
Comment (z-sztrom) [reply publish-cli/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: removed

File: publish-service/src/test/java/com/ericsson/eiffel/remrem/publish/service/EventTemplateHandlerTest.java [hunk 15]
```diff
@@ -333,50 +330,47 @@ public void testEventParserWithEventEiffelServiceStoppedEvent() {
...
             ObjectMapper mapper = new ObjectMapper();
             JsonNode expectedJson = mapper.readTree(expectedDocument);
-
             JsonNode actualParsedEventJson = eventTemplateHandler.eventTemplateParser(dataToBeParsed, EventName);
 
             LOG.info("expectedJsonString: " + expectedJson.toString());
             LOG.info("actualParsedEventJson: " + actualParsedEventJson.toString());
 
-            JSONAssert.assertEquals(expectedJson.toString(), actualParsedEventJson.toString(), true);
+            JSONAssert.assertEquals(expectedJson.toString(), actualParsedEventJson.toString(), false);
```
[the remark this answers is not in the document]
Comment (z-sztrom) [reply publish-service/src/test/java/com/ericsson/eiffel/remrem/publish/service/EventTemplateHandlerTest.java:368]: I don't know... Restored back to 'true'.

**Review by jainadc9 — APPROVED** [review]

**Pull request merged**